packages/cli/src/tools/assign.ts

```diff
@@ -4,14 +4,16 @@
  * Similar to Object.assign(), but it doesn't execute getters. This allows us to have
  * lazy properties on an object and still be able to merge them together
  *
- * @flow
  */
 export default function assign(target: Object, ...sources: Object[]) {
   sources.forEach(source => {
-    let descriptors = Object.keys(source).reduce((acc, key) => {
-      acc[key] = Object.getOwnPropertyDescriptor(source, key);
-      return acc;
-    }, {});
+    let descriptors = Object.keys(source).reduce(
+      (acc: {[k: string]: any}, key) => {
```

Comment:
how about moving the type to initial value in second argument of reduce? also, please use "unknown" instead of "any"

Reply:
May `PropertyDescriptor` it's a better fit than `unknown`?

```
(acc: {[key: string]: PropertyDescriptor | undefined}, key: string) => {
```

Reply:
New update using `PropertyDescriptorMap`. I had to add a safety check because `getOwnPropertyDescriptor` also returns undefined.

Reply:
Also `uknown` does not work that great here because you cannot pass it to `Object.defineProperties` later on.